apps/builder/app/builder/shared/sync/sync-server.ts

```diff
@@ -125,18 +303,18 @@ export const useSyncServer = ({
   authPermit,
   version,
 }: SyncServerProps) => {
-  // save the initial version from loaded build
-  const initialized = useRef(false);
-  useEffect(() => {
-    if (authPermit === "view" || initialized.current === true) {
-      return;
-    }
-    initialized.current = true;
-    syncServer({ buildId, projectId, authToken, lastVersion: version });
-  }, [buildId, projectId, authToken, authPermit, version]);
+  useSyncProject({
```

Comment:
Why not export useSyncProject directly?

Reply:
to combine with `useBeforeUnload` and not expose this above